protocol/contracts/core/MultiOtcOffer.sol

```diff
@@ -0,0 +1,187 @@
+    
+    }
+
+    function removeOtcMarket(bytes32 _marketId) external override onlyElevatedAccess {
+        if (!_otcMarketIds.remove(_marketId)) { revert InvalidMarketId(_marketId); }
+        /// cache for events
+        OTCMarketInfo memory marketInfo = otcMarketInfo[_marketId];
+        address userBuyToken = address(marketInfo.userBuyToken);
+        address userSellToken = address(marketInfo.userSellToken);
+        delete otcMarketInfo[_marketId];
+        delete otcMarketPriceParams[_marketId];
+
+        emit OtcMarketRemoved(_marketId, userBuyToken, userSellToken);
+    }
+
+    function setMarketFundsOwner(bytes32 _marketId, address _fundsOwner) external override onlyElevatedAccess {
+        if (!_otcMarketIds.contains(_marketId)) { revert InvalidMarketId(_marketId); }
+        if (_fundsOwner == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        OTCMarketInfo storage _otcMarketInfo = otcMarketInfo[_marketId]; 
+        _otcMarketInfo.fundsOwner = _fundsOwner;
+        emit FundsOwnerSet(_marketId, _fundsOwner);
+    }
+
+    function setOfferPrice(bytes32 _marketId, uint256 _offerPrice) external override onlyElevatedAccess {
+        if (!_otcMarketIds.contains(_marketId)) { revert InvalidMarketId(_marketId); }
+        OTCMarketPriceParams storage marketPriceParam = otcMarketPriceParams[_marketId];
+        if (_offerPrice < marketPriceParam.minValidOfferPrice || _offerPrice > marketPriceParam.maxValidOfferPrice) revert OfferPriceNotValid();
+        marketPriceParam.offerPrice = _offerPrice;
+        emit OfferPriceSet(_marketId, _offerPrice);
+    }
+
+    function setOfferPriceRange(bytes32 _marketId, uint128 _minValidOfferPrice, uint128 _maxValidOfferPrice) external override onlyElevatedAccess {
+        if (_minValidOfferPrice > _maxValidOfferPrice) revert CommonEventsAndErrors.InvalidParam();
+        OTCMarketPriceParams storage marketPriceParam = otcMarketPriceParams[_marketId];
+        marketPriceParam.minValidOfferPrice = _minValidOfferPrice;
+        marketPriceParam.maxValidOfferPrice = _maxValidOfferPrice;
+        emit OfferPriceRangeSet(_marketId, _minValidOfferPrice, _maxValidOfferPrice);
+    }
+
+    /// @notice Owner can pause user swaps from occuring
+    function pause() external override onlyElevatedAccess {
+        _pause();
+    }
+
+    /// @notice Owner can unpause so user swaps can occur
+    function unpause() external override onlyElevatedAccess {
+        _unpause();
+    }
+
+    /// @notice Swap `userSellToken` for `userBuyToken`, at the `offerPrice`
+    function swap(bytes32 marketId, uint256 sellTokenAmount) external override whenNotPaused returns (uint256 buyTokenAmount) {
+        if (!_otcMarketIds.contains(marketId)) { revert InvalidMarketId(marketId); }
+        if (sellTokenAmount == 0) revert CommonEventsAndErrors.ExpectedNonZero();
+        OTCMarketInfo memory marketInfo = otcMarketInfo[marketId];
+        buyTokenAmount = quote(marketId, sellTokenAmount);
```

Comment:
Calling quote here is doing more than it has to -- eg `_otcMarketIds.contains` is being called twice. I'd move the mulDiv into a separate internal function for both to use